### PR TITLE
KUNST-72: The big upgrade

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/KUNST-72

Upgrades to 

* PHP 8.1
* Symfony 6.3

Assets, i.e. Webpack and the like, are not yet upgraded.

The commit history is a mess and we should rebase into a single commit if this is ever approved (or squash when merging).

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
+# DATABASE_URL="sqlite:///%kernel.project_dir%/var/data.db"
+# DATABASE_URL="mysql://app:!ChangeMe!@127.0.0.1:3306/app?serverVersion=8.0.32&charset=utf8mb4"
+# DATABASE_URL="mysql://app:!ChangeMe!@127.0.0.1:3306/app?serverVersion=10.11.2-MariaDB&charset=utf8mb4"
+# DATABASE_URL="postgresql://app:!ChangeMe!@127.0.0.1:5432/app?serverVersion=15&charset=utf8"
 DATABASE_URL=mysql://db:db@mariadb:3306/db
-# Override to use mariadb as backend
-DATABASE_SERVER_VERSION='mariadb-10.3.13'
 ###< doctrine/doctrine-bundle ###
 
 APP_SECRET=NotSoSecretReplace
```
Comment (tuj) [inline .env:36]: APP_SECRET defined two times in the file

File: README.md [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+WEB_ACCESSIBILITY_STATEMENT_URL=…
+
+# Install
+itkdev-docker-compose-server up --detach --remove-orphans
+itkdev-docker-compose-server exec phpfpm composer install --no-dev
+itkdev-docker-compose-server exec phpfpm bin/console doctrine:migrations:migrate --no-interaction
+
+# Build assets using our development setup
+docker compose run --rm node yarn install
+docker compose run --rm node yarn build
```
Comment (tuj) [inline README.md:34]: and remove the node_modules folder

File: migrations/Version20230801120357.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // The migration_versions table exists only if upgrading, i.e. not making a fresh install.
+        $this->addSql('DROP TABLE IF EXISTS migration_versions');
```
Comment (tuj) [inline migrations/Version20230801120357.php:23]: Why is this needed?

**Review by tuj — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (rimi-itk) [reply .env]: Fixed in https://github.com/itk-dev/kunstdatabasen/pull/68/commits/f734d8617b9accb1914b2ec4d9eff2c67f9b5c29.

File: README.md (hunk 2 above)
Comment (rimi-itk) [reply README.md:38]: Fixed in https://github.com/itk-dev/kunstdatabasen/pull/68/commits/84c646607fc9507b278089e70abe0c5bc243a2ba.

File: migrations/Version20230801120357.php (hunk 3 above)
Comment (rimi-itk) [reply migrations/Version20230801120357.php:28]: I've updated the comment above the line with more details and a reference to https://github.com/doctrine/DoctrineMigrationsBundle/blob/3.2.x/UPGRADE.md#from-2x-to-300 (cf. https://github.com/itk-dev/kunstdatabasen/pull/68/commits/5a80bb354723625d0ca6482cd67f826ec3280a76).

How de we usually handle this situation? Do we add configuration to keep the `migration_versions` table name?

Comment (tuj) [reply migrations/Version20230801120357.php:28]: "and existing database" => "an existing database"

**Review by tuj — APPROVED** [review]

**Pull request merged**